src/lib/NetworkClient.js

```diff
@@ -149,13 +149,30 @@ class NetworkClient {
      * @returns {Promise<Array<String>|Error>}
      */
     async getEssidList(iface) {
+        const essid_label = 'ESSID';
         const link_up = `ip link set ${iface} up`;
         const scan = `iwlist ${iface} scan`;
-        const grep_and_cut_essid = "/usr/bin/grep ESSID | /usr/bin/cut -d':' -f2 | /usr/bin/cut -d'\"' -f2";
         const output = await cockpit.spawn(
-            ["bash", "-c", `${link_up} && ${scan} | ${grep_and_cut_essid}`], { superuser: true }
+            ["bash", "-c", `${link_up} && ${scan}`], { superuser: true }
         );
-        return output.trim().split("\n");
+
+        const extractValue = function(label_value_pair) {
+            const quoted_value = label_value_pair
```

Comment:
Please, keep naming consistent (`quotedValue` instead of `quoted_value`). I know, the original code did not respect this convention, but now that we are rewriting it... If you prefer, I may take care after merging.

Reply:
I've changed this now, though maybe it makes it more difficult to read since `_` added spaces between words.